### DISCUSSION TITLE
How to dynamic import scss

### DISCUSSION THREAD

**Discussion by Capricair** [vercel/next.js#47058] [Help]
### Summary

I want to add some theme to my website using css variables, and dynamic import a scss file in _app.js according to the theme data returned by the api, what should i do?

I try to do it like this

```
if (theme === "a") {
   require("styles/themes/a.scss")
} else if (theme === "b") {
  require("styles/themes/b.scss")
}
```

but after next build, theme a and b are both imported

next/dynamic must return a class/function, cannot import scss

**Comment by icyJoseph** [comment]
For this use case, you could go with `import('styles/theme/a.css')`, but the tricky part is that you need to unmount the style when you don't need it anymore... 

For example there's a Community Driven, Webpack plugin which has a style loader, that exposes the `use/unuse` API - Maybe you can use that in your next.config.js:  https://webpack.js.org/loaders/style-loader/#lazystyletag

This page uses it for CSR: https://mertsincan.github.io/primereact-dynamic-theming/

You could also just try to inject the new styles constantly, but that feels wrong (like light->dark->light would have to append the light styles twice).
